File: Template Method/Golang/TwoFactorEmail.go
package main

import "fmt"

type TwoFactorEmail struct {}

func (twoFactor *TwoFactorEmail) getData(username string)  {
	fmt.Println("Getting email data for", username)
}

func (twoFactor *TwoFactorEmail) sendNotification(username string)  {
	fmt.Println("Sending email to", username)
}

func (twoFactor *TwoFactorEmail) match(username string)  {
	fmt.Println("Matching email code from", username)
}

File: Template Method/Python 3/TwoFactorEmail.py
from TwoFactorAuthentication import TwoFactorAuthentication

class TwoFactorEmail(TwoFactorAuthentication):
  def getData(self, username):
    print('Getting email data for {}'.format(username))

  def sendNotification(self, username):
    print('Sending email notification to {}'.format(username))

  def match(self, username):
    print('Matching {} with the email code'.format(username))